src/pages/page-plan/page-plan.tsx

```diff
@@ -1,159 +0,0 @@
-import { useState, type FC } from "react";
-// components
-import { PagePlanProps } from "./page-plan.interface";
-import { useFormContext } from "react-hook-form";
-import { FormTypes } from "../../types/form-types";
-import { InputPlanCard } from "../../components/input-plan-card/input-plan-card";
-//hooks
-import { useDeviceType } from "../../hooks/use-device-type";
-// assets
-import arcade from "../../assets/images/icon-arcade.svg";
-import advanced from "../../assets/images/icon-advanced.svg";
-import pro from "../../assets/images/icon-pro.svg";
-import { ToogleSwitch } from "../../components/toogle-switch/toogle-switch";
-
-export const PagePlan: FC<PagePlanProps> = ({
-  onBack,
-  onContinue,
-  isMonthly,
-  toggleBilling,
-}) => {
-  const deviceType = useDeviceType();
-  // state
-  const [selectedPlan, setSelectedPlan] = useState("");
-
-  // form
-  const {
-    register,
-    trigger,
-    formState: { errors },
-    setValue,
-  } = useFormContext<FormTypes>();
-
-  const handlePlanChange = async (
-    e: React.ChangeEvent<HTMLInputElement>,
-    price: string
-  ) => {
-    e.preventDefault();
-    const value = e.target.value;
-
-    setSelectedPlan(value);
-    setValue("plan", { value, price, isMonthly });
-    trigger("plan");
-  };
-
-  const handleBack = () => {
-    setValue("plan", {
-      value: "",
-      price: "",
-      isMonthly: true,
-    });
-
-    onBack();
-  };
-
-  const handleContinue = async (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    const isValid = await trigger(["plan"]);
-
-    if (isValid) {
-      onContinue();
-    }
-  };
-
-  return (
-    <form
-      onSubmit={handleContinue}
-      className="bg-white h-full  flex flex-col justify-between w-full rounded-lg shadow-xl md:shadow-none md:pt-10 py-9 md:py-6 px-7 md:px-12"
-    >
-      <div className="flex flex-col gap-4">
-        <h2 className="text-marine text-3xl font-semibold">Select your plan</h2>
-        <p className="text-cool-gray leading-5">
-          You have the option of monthly or yearly billing.
-        </p>
-        <div className="flex flex-col md:pt-7 gap-3 md:gap-5 md:flex-row">
-          <InputPlanCard
-            img={arcade}
-            alt="arcade"
-            title="Arcade"
-            price={isMonthly ? "$9/mo" : "$90/ur"}
-            value="Arcade"
-            checked={selectedPlan === "Arcade"}
-            isMonthly={isMonthly}
-            {...register("plan")}
-            onChange={(e) =>
-              handlePlanChange(e, isMonthly ? "$9/mo" : "$90/ur")
```

Comment:
It's better to create a constant array for the plan list. Here's an example:

```ts
export type Plan = {
  id: string;
  title: string;
  monthPrice: number; // cents
  yearPrice: number; // cents
}

export const plans: Plan[] = [{
  id: 'arcade',
  title: 'Arcade',
  monthPrice: 900,
  yearPrice: 9000
}, {
  id: 'advanced'
  ...
}]

{plans.map((plan) => {
  return (
    <InputPlanCard 
      key={plan.id} 
      title={plan.title} 
      price={formatCents(isMonthly ? plan.monthPrice : plan.yearPrice)}
      ... 
    />
  );
})}
```

The same logic can be applied to the addons list.

In the form, you can store only the plan id and array of addons ids. To display the plan details later, you can search for the corresponding plan in the plans array.

```ts
const { pickOns, plan } = watch();  // pickOns = string[], plan = string

const totalPrice = useMemo(() => {
  const selectedPlan = plans.find(({ id }) => id === plan);
  const selectedAddons = addons.filter(({ id }) => pickOns.includes(id));

  const planPrice = isMonthly ? selectedPlan.monthPrice : selectedPlan.yearPrice;
  const addonsPrice = selectedAddons.reduce((sum, current) => {
    return sum + isMonthly ? current.monthPrice : current.yearPrice;
  }, 0)
  
  return planPrice + addonsPrice;
}, [...])
```